modules/test-structure/save_test_data.go

```diff
@@ -282,3 +282,28 @@ func CleanupTestData(t *testing.T, path string) {
 		logger.Logf(t, "%s does not exist. Nothing to cleanup.", path)
 	}
 }
+
+// CleanupTestDataFolder cleans up the .test-data folder inside the given folder.
+// If deleteParent is true, CleanupTestDataFolder will also delete the given folder.
+func CleanupTestDataFolder(t *testing.T, path string, deleteParent bool) {
```

Comment:
Hm, `deleteParent` seems like a strange, and potentially dangerous, thing to toss into the API. What's the use case for that?

Reply:
Currently, I am using the `CopyTerraformFolderToTemp` and setting the output path as my working directory for all the `Save*` functions. I do this to ensure that all my parallel tests init with their own config. After the tests, I would like to clean up the multiple copies (using the `deleteParent` flag).

There are other situations where I do not use `CopyTerraformFolderToTemp` and thus I would only want to delete the `.test-data` (using `false` for `deleteParent`) folder inside the working directory that I passed to the `Save*` functions.

Would be happy to hear if there are better ways to achieve my use case. Thanks! :smile:

Reply:
I think this is mixing two use cases into one function, which is probably not the right approach. `CleanupTestDataXXX` should only deal with the test data stuff in `.test-data`. Cleaning up `CopyTerraformFolderToTemp` contents should be done via a separate function dedicated to that use case. Mixing the two together is confusing and, given that we are talking about deleting whole folders of content, it's too risky to do anything that isn't crystal clear and obvious.

Reply:
Sure that works too. I also realised that with the `deleteParent` flag, the function is more or less just a `os.RemoveAll` call. I will make the necessary changes.Thanks!